typing/mode.ml

```diff
@@ -2301,38 +2317,34 @@ module Value_with (Areality : Areality) = struct
   type lr = (allowed * allowed) t
 
   module Axis = struct
-    type ('a, 'd0, 'd1) t =
-      | Monadic : (Monadic.Const.t, 'a) Axis.t -> ('a, 'd, 'd neg) t
-      | Comonadic : (Comonadic.Const.t, 'a) Axis.t -> ('a, 'd, 'd pos) t
+    type 'a t =
+      | Monadic : 'a Monadic.Axis.t -> 'a t
+      | Comonadic : 'a Comonadic.Axis.t -> 'a t
 
-    let compare : type a d0 d1 b e0 e1. (a, d0, d1) t -> (b, e0, e1) t -> int =
+    let compare : type a b. a t -> b t -> int =
      fun t0 t1 ->
       match t0, t1 with
       | Monadic t0, Monadic t1 -> Axis.compare t0 t1
-      | Monadic _, Comonadic _ -> 1
-      | Comonadic _, Monadic _ -> -1
+      | Monadic t0, Comonadic t1 -> Axis.compare t0 t1
```

Comment:
What is happening in those two lines? Why are we comparing monadic and comonadic axes?